test/TestUtilities/GraphicsUtil.hpp

```diff
@@ -38,3 +42,66 @@ bool operator==(const sf::Rect<T>& lhs, const Approx<sf::Rect<T>>& rhs)
 {
     return lhs.position == Approx(rhs.value.position) && lhs.size == Approx(rhs.value.size);
 }
+
+template <typename T>
+struct EqualsApproxVector2Matcher : Catch::Matchers::MatcherGenericBase
```

Comment:
Can you talk about it about how `Approx(vec)` was not sufficient and why you needed to add this new helper class template?

Reply:
Floating point error accumulates the more operations you perform. Especially when checking metrics values over longer strings of text, the error/rounding that is done to get to that point goes beyond the standard epsilon value of `Approx()` which is less than 1 pixel. Strict equality comparison like it was done previously does not work any more since shaping can and will return sub-1-pixel positions. If we were to use `Approx()` with `.margin(1)`, it would have also needed an overload for comparing `sf::Vector2f`s and `sf::FloatRect`s. Also, according to [Catch2's own documentation](https://github.com/catchorg/Catch2/blob/devel/docs/comparing-floating-point-numbers.md#approx) using `Approx()` is not recommended for newer code.